tests/e2e/streams/common.py

```diff
@@ -129,28 +131,25 @@ def validate_info(actual_stream_info, expected_stream_info):
     for info, expected_info in zip(actual_stream_info, expected_stream_info):
         assert info == expected_info
 
+
 def check_stream_info(cursor, stream_name, expected_stream_info):
     stream_info = get_stream_info(cursor, stream_name)
     validate_info(stream_info, expected_stream_info)
 
+
 def kafka_check_vertex_exists_with_topic_and_payload(cursor, topic, payload_bytes):
-    decoded_payload = payload_bytes.decode('utf-8')
-    check_vertex_exists_with_properties(
-            cursor, {'topic': f'"{topic}"', 'payload': f'"{decoded_payload}"'})
+    decoded_payload = payload_bytes.decode("utf-8")
+    check_vertex_exists_with_properties(cursor, {"topic": f'"{topic}"', "payload": f'"{decoded_payload}"'})
 
 
-PULSAR_SERVICE_URL = 'pulsar://127.0.0.1:6650'
+PULSAR_SERVICE_URL = "pulsar://127.0.0.1:6650"
+
 
 def pulsar_default_namespace_topic(topic):
-    return f'persistent://public/default/{topic}'
+    return f"persistent://public/default/{topic}"
 
 
-def test_start_and_stop_during_check(
-        operation,
-        connection,
-        stream_creator,
-        message_sender,
-        already_stopped_error):
+def test_start_and_stop_during_check(operation, connection, stream_creator, message_sender, already_stopped_error):
```

Comment:
I think it is easy to parameterize this test to call `CHECK STREAM` and `START STREAM .. LIMIT_BATCHES`. Maybe you can give it a try.

Reply:
Test `test_start_and_stop_during_check` seems quite complex already, I am wondering/afraid parameterizing it would make it even more complex and hard to maintain.
I can still give it a try however. What's your call on that?